### PR TITLE
Feature: advanced Open Orders from Finex API

### PR DESCRIPTION
**Pull request by josadcha**

### CODE REVIEW

**Review by akushniruk — APPROVED** [review]

File: web/src/containers/OpenOrders/index.tsx [hunk 1]
```diff
@@ -1,180 +1,164 @@
...
-        } else if (userLoggedIn && currentMarket && prevCurrentMarket !== currentMarket) {
-            this.props.userOpenOrdersFetch({ market: currentMarket });
+    if (ordType.includes('take')) {
+        if (ordSide === 'buy') {
+            return '≤';
+        } else {
+            return '≥';
         }
     }
+}
```
Comment (akhlopiachyi) [inline web/src/containers/OpenOrders/index.tsx:36]: maybe better to create helpers.ts inside web/src/containers/OpenOrders folder and move this function there?

File: web/src/containers/OpenOrders/index.tsx [hunk 2]
```diff
@@ -1,180 +1,164 @@
...
+        translate('page.body.trade.header.openOrders.content.price'),
+        translate('page.body.trade.header.openOrders.content.amount'),
+        translate('page.body.trade.header.openOrders.content.total'),
+        translate('page.body.trade.header.openOrders.content.trigger'),
+        translate('page.body.trade.header.openOrders.content.filled'),
+        '',
+    ];
+
+    const renderData = useMemo(() => {
         if (list.length === 0) {
```
Comment (akhlopiachyi) [inline web/src/containers/OpenOrders/index.tsx:76]: !list.length

File: web/src/containers/OpenOrders/index.tsx [hunk 3]
```diff
@@ -1,180 +1,164 @@
...
-    connect(
-        mapStateToProps,
-        mapDispatchToProps,
-    )(OpenOrdersContainer),
-) as React.FunctionComponent;
+    const classNames = useMemo(() => 
+        classnames('pg-open-orders', {
+            'pg-open-orders--empty': !list.length,
+            'pg-open-orders--loading': fetching,
+        }),
```
Comment (akhlopiachyi) [inline web/src/containers/OpenOrders/index.tsx:124]: move `),` to next line

File: web/src/containers/OpenOrders/index.tsx [hunk 4]
```diff
@@ -1,180 +1,164 @@
...
+            </div>
+            {fetching && <div className="open-order-loading"><Spinner animation="border" variant="primary" /></div>}
+            {!fetching && (
+                <OpenOrders
+                    headersKeys={headersKeys}
+                    headers={renderHeaders}
+                    data={renderData}
+                    onCancel={handleCancel}
+                />
+            )}
```
Comment (akhlopiachyi) [inline web/src/containers/OpenOrders/index.tsx:161]: Please create a method using useMemo that returns html depends on `fetching` value

File: web/src/modules/user/openOrders/actions.ts [hunk 5]
```diff
@@ -15,7 +15,7 @@ import {
 
 export interface UserOpenOrdersFetch {
     type: typeof OPEN_ORDERS_FETCH;
-    payload: {
+    payload?: {
         market: Market,
```
Comment (akhlopiachyi) [inline web/src/modules/user/openOrders/actions.ts:19]: please replace `market: Market,` with `market: Market;`

File: web/src/containers/OrdersElement/index.tsx [hunk 6]
```diff
@@ -143,33 +146,63 @@ class OrdersComponent extends React.PureComponent<Props, OrdersState>  {
...
+
+    private getPrice = (ord_type, status, avg_price, trigger_price, price) => {
+        if (ord_type === 'market' || status === 'done') {
+            return avg_price;
+        } else if (status === 'trigger_wait') {
+            return trigger_price;
+        } else {
+            return price;
+        }
     };
```
Comment (akhlopiachyi) [inline web/src/containers/OrdersElement/index.tsx:206]: I guess be better to create a helpers file for OrdersElement and move this function there

**Review by akhlopiachyi — APPROVED** [review]

File: web/src/containers/OpenOrders/index.tsx [hunk 7]
```diff
@@ -1,180 +1,157 @@
...
-                <span style={{ color: setTradeColor(side).color }} key={id}>{Decimal.format(+remaining_volume, amountFixed, ',')}</span>,
-                <span style={{ color: setTradeColor(side).color }} key={id}>{filled}%</span>,
+                <span key={id} className="split-lines"><span className="secondary">{localeDate(created_at, 'date')}</span>&nbsp;<span>{localeDate(created_at, 'time')}</span></span>,
+                <span key={id} className="bold">{curMarket?.name.toUpperCase()}</span>,
+                <span key={id}>{ord_type ? translate(`page.body.trade.header.openOrders.content.type.${ord_type}`) : '-'}</span>,
+                <span style={{ color: setTradeColor(side).color }} key={id}><Decimal fixed={priceFixed} thousSep=",">{price}</Decimal></span>,
+                <span key={id}><Decimal fixed={amountFixed} thousSep=",">{+remaining_volume}</Decimal></span>,
+                <span key={id}><Decimal fixed={amountFixed} thousSep=",">{+remaining_volume}</Decimal> <span className="cr-text__opacity">{curMarket?.quote_unit?.toUpperCase()}</span></span>,
```
Comment (akushniruk) [inline web/src/containers/OpenOrders/index.tsx:77]: multiply by price

File: web/src/components/OrderForm/index.tsx [hunk 8]
```diff
@@ -523,8 +523,13 @@ export class OrderForm extends React.PureComponent<OrderFormProps, OrderFormStat
 
     private handleChangeAmountByButton = (value: number) => {
         const { orderType, price, trigger } = this.state;
+        const ordType = (orderType as string).toLowerCase()
 
-        const priceToUse = (orderType as string).toLowerCase().includes('limit') || orderType === 'Market' ? price : trigger;
+        let priceToUse = ordType.includes('limit') || orderType === 'Market' ? price : trigger;
+
+        if (['stop-loss', 'take-profit'].includes(ordType)) {
+            priceToUse = (Number(priceToUse) * 1.1).toString()
```
Comment (akushniruk) [inline web/src/components/OrderForm/index.tsx:531]: pls put 1.1 to the constants file

**Review by akushniruk — CHANGES_REQUESTED** [review]

**Review by akushniruk — APPROVED** [review]

**Pull request merged**